slapp/transforms/image_utils.py

```diff
@@ -1,76 +1,56 @@
-from typing import Tuple, Union
-from pathlib import Path
-from math import ceil
-
+from typing import Tuple
 import numpy as np
-from PIL import Image, ImageDraw, ImageFont
+import cv2
 
 
-def add_scale(image: Union[Path, np.ndarray],
+def add_scale(array: np.ndarray,
               scale_position: Tuple[int, int],
-              font_file: Path,
-              resolution: float = 400 / 512,
-              scale_size: int = 10,
-              fill: Tuple[int, int, int] = (0, 0, 0),
-              width: int = 5,
-              font_size: int = 8) -> Image:
+              um_per_pixel: float = 400 / 512,
+              scale_size_um: float = 10,
+              color: int = 1,
+              thickness_um: float = 1.0,
+              fontScale: float = 0.2) -> np.ndarray:
     """
-    Adds a scale bar onto a image passed in the form of a image path or
-    a numpy array. Uses PIL library ImageDraw and ImageFont to draw scale bar
-    onto images.
+    Adds a scale bar onto an array using opencv.
+
     Parameters
     ----------
-    image: [Path, np.ndarray]
-        path to the image to add scale bar
+    array: np.ndarray
+        function only accepts np.uint8 2D arrays
     scale_position: Tuple
-        the position at which to set the bottom left corner of the scale. In
-        x, y format, this is the format PIL expects.
-    font_file: Path
-        the font file to be used as denoted by a path object. On windows can
-        just be specified as 'arial.ttf' on linux must be a full path.
-    resolution: float = 10/13
-        resolution of the video in microns per pixel, defaulted to
-        scientifica general which is 10/13~.78 micros/pixel
-    scale_size: int=10
-        how many micro meters the scale bar should display, defaulted to 10
-    fill: Tuple=(0, 0, 0)
-        This parameter is passed to line draw in PIL libary and name reflects
-        name in that API. The fill color in rgb tuple to draw the text and
-        scale, defaults to black
-    width: Int
-        This parameter is passed to line draw in PIL libary and name reflects
-        name in that API. The width of the lines of the scale in pixels
-    font_size: Int
-        the font size of the description text
+        image coordinates for the corner of the scale bars
+    um_per_pixel: float = 400/512
+        resolution of the video in microns per pixel
+    scale_size_um: float
+        length in microns of scale bars
+    color: int
+        default 1. passed as `color` to cv2.line and cv2.putText
+    thickness_um: float
```

Comment:
Nit: Why do the line thickness in microns? Pixels makes more sense to me since the line thickness has nothing to do with the resolution.

Reply:
the only reason is that the `scale_size` is in um, which makes sense since we are annotating in um and providing `um_per_pixel`. With both `scale_size` and `thickness` in um, we can understand the aspect ratio of the line we are about to draw.
Your way makes sense also. I am happy to change it.

Reply:
I can see both sides. I personally would prefer pixels because I'm not thinking about the aspect ratio as much as I am the actual thickness of the line in pixels on the screen. But that's why this is a nit. I'll leave it up to you.

Reply:
Will leave in um.